File: 1ºAno/Lista3/Python/Sala/files/ex5.py
while True:
    while True:
        try:
            div, index = input("Na mesma linha, entre com os valores, respectivamente, referentes a:\n\n1) Índice de divisibilidade\n2) Número de parada\nPara sair, entre com 0 em todos os valores\n\n").split()
            div, index = int(div), int(index)
            break
        except:
            print ("Número(s) inválido(s)... Tente novamente")
    if div == index == 0:
        break
    print ("Aqui estão os números divisíveis por {} que são menores ou iguais a {}:" .format(div, index))
    for i in range (div, index + 1, div):
        print ("{} " .format(i), end='')
    print ("\n")
print ("\nAdeus!")

File: 1ºAno/Lista3/C/Sala/files/ex5.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main() {
    for (;;) {
        char *dados = (char*)malloc(20 * sizeof (char));
        int val = 0, index = 0;
        for (;;) {
            printf ("Entre, na mesma linha, com os valores, respectivamente, referentes a:\n\n1) Valor de divisibilidade\n2) Índice do programa\n\n");
            fgets (dados, 20, stdin);
            char *pt = strtok(dados, " ");
            val = atoi(pt);
            pt = strtok(NULL, " ");
            index = atoi(pt);
            if (val == 0 && index != 0 || index == 0 && val != 0) {
                printf ("*Número(s) inválido(s)... Tente novamente*\n");
            }
            else {
                break;
            }
        }
        if (val == 0 && index == 0) {
            break;
        }
        printf ("Aqui estão os números n divisíveis por %d, tal que n < %d:\n", val, index);
        int cont = 1;
        for (cont = val; cont <= index; cont += val) {
            printf ("%d ", cont);
        }
        printf ("\n\n");
    }
    printf ("\nAdeus!");
    return 0;
}